api/v1/submissions/PKPSubmissionController.php

```diff
@@ -1558,6 +1559,15 @@ public function addContributor(Request $illuminateRequest): JsonResponse
 
         $author = Repo::author()->newDataObject($params);
         $newId = Repo::author()->add($author);
+
+        $params['id'] = $newId;
+        $author->setId($newId);
+        $author->setAffiliations($params['affiliations']);
```

Comment:
is $params['affiliations'] array of Affiliation objects?

Reply:
I think you should here do:
1. Repo::affiliation()->validate($author, $params, $submission, $submissionContext);
2. if error -> return
3. Repo::affiliation()->newDataObject($params);

Then no need to have author->setAffiliations, because you get the author at the end `$author = Repo::author()->get($newId);` which will get the affiliations.

Reply:
Maybe also to do here:
foreach ($params['affiliations'] as $affiliationParam) {
then the steps above
and having `Repo::affiliation()->validate(null, $params, $author, $submission, $submissionContext);` i.e. validating only one affiliation i.e. having the first parameter Affiliation object -- see below where the validate is implemented.

Reply:
Maybe to create Affiliation objects and use author->setAffiliations before adding the author.
Then in the author DAO the author is first added and then the affiliations saved, containing the author ID.